source/sn_coap_parser.c

```diff
@@ -380,7 +379,7 @@ static int8_t sn_coap_parser_options_parse(struct coap_s *handle, uint8_t **pack
                 /* This is managed independently because User gives this option in one character table */
 
                 ret_status = sn_coap_parser_options_parse_multiple_options(handle, packet_data_pptr,
-                             message_left,
+                             packet_len - ((*packet_data_pptr) - packet_data_start_ptr) + 1,
```

Comment:
Is there a reason you are re-calculating the proper value for message_length in few places instead of trying to keep it up to date on lines 275 and 540? The message_left is used as loop termination condition so I believe it should have correct value.